src/drivers/intel/hda/hda-dma.c

```diff
@@ -630,6 +629,28 @@ static int hda_dma_stop(struct dma_chan_data *channel)
 	return 0;
 }
 
+static int hda_dma_reset(struct dma_chan_data *channel)
+{
+	uint32_t val, flags;
+
+	irq_local_disable(flags);
+
+	/* reset GEN bit */
+	dma_chan_reg_update_bits(channel, DGCS, DGCS_GEN, 0);
+
+	val = dma_chan_reg_read(channel, DGCS);
+
+	irq_local_enable(flags);
+
+	/* and verify if channel is idle */
+	if (!(val & DGCS_GBUSY))
```

Comment:
Do we need some timeout check here?

Reply:
@keyonjie I checked this in the cavs fw and they do not have a wait before checking for GBUSY